cpp/src/arrow/compute/internals_benchmark.cc

```diff
@@ -0,0 +1,86 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
We already have `function_benchmark.cc` for such internals benchmarks.

Reply:
I can merge these together then.